pkgs/development/libraries/qt-5/modules/qtwebengine.nix

```diff
@@ -30,46 +29,41 @@
 , pipewireSupport ? stdenv.isLinux
 , pipewire_0_2
 , postPatch ? ""
-, nspr
-, lndir
 , dbusSupport ? !stdenv.isDarwin, expat
 }:
 
 let
-  # qtwebengine expects to find an executable in $PATH which runs on
-  # the build platform yet knows about the host `.pc` files.  Most
-  # configury allows setting $PKG_CONFIG to point to an
-  # arbitrarily-named script which serves this purpose; however QT
-  # insists that it is named `pkg-config` with no target prefix.  So
-  # we re-wrap the host platform's pkg-config.
-  pkg-config-wrapped-without-prefix = stdenv.mkDerivation {
-    name = "pkg-config-wrapper-without-target-prefix";
-    dontUnpack = true;
-    dontBuild = true;
-    installPhase = ''
-      mkdir -p $out/bin
-      ln -s '${buildPackages.pkg-config}/bin/${buildPackages.pkg-config.targetPrefix}pkg-config' $out/bin/pkg-config
+  isCrossBuild = stdenv.buildPlatform != stdenv.hostPlatform;
+
+  # qtwebengine requires its own very particular specially patched for qt version of gn
+  gnQtWebengine = with srcs.qtwebengine; buildPackages.gn.overrideAttrs {
```

Comment:
If this thing is too scary, I can just remove it, but there will be one less binary shared between native and cross builds.